enterprise/server/util/cacheproxy/cacheproxy_test.go

```diff
@@ -131,6 +131,56 @@ func TestReader(t *testing.T) {
 	}
 }
 
+func TestReaderMaxOffset(t *testing.T) {
+	ctx := context.Background()
+	flags.Set(t, "auth.enable_anonymous_usage", "true")
+	te := getTestEnv(t, emptyUserMap)
+
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, te)
+	if err != nil {
+		t.Errorf("error attaching user prefix: %v", err)
+	}
+
+	peer := fmt.Sprintf("localhost:%d", app.FreePort(t))
+	c := cacheproxy.NewCacheProxy(te, te.GetCache(), peer)
+	if err := c.StartListening(); err != nil {
+		t.Fatalf("Error setting up cacheproxy: %s", err)
+	}
+	waitUntilServerIsAlive(peer)
+
+	randomSrc := &randomDataMaker{rand.NewSource(time.Now().Unix())}
+
+	// Read some random bytes.
+	buf := new(bytes.Buffer)
+	io.CopyN(buf, randomSrc, 100)
+	readSeeker := bytes.NewReader(buf.Bytes())
+
+	// Compute a digest for the random bytes.
+	d, err := digest.Compute(readSeeker)
+	if err != nil {
+		t.Fatal(err)
+	}
+	readSeeker.Seek(0, 0)
+
```

Comment:
nit: set instanceName := "foo" and use that variable below?

Reply:
Done